ydb/core/tx/datashard/datashard__read_iterator.cpp

```diff
@@ -2657,6 +2657,16 @@ void TDataShard::Handle(TEvDataShard::TEvRead::TPtr& ev, const TActorContext& ct
         return;
     }
 
+    if (State == TShardState::PreOffline ||
+        State == TShardState::Offline)
+    {
+        replyWithError(
+            Ydb::StatusIds::NOT_FOUND,
```

Comment:
Don't you want to catch is the test  `EvReadResult` and check status 'NOT_FOUND' ?

Reply:
Why? I want to observe and test the effect (high latency), not some implementation detail.